src/parser/open-api/v3/open-api-v3-parser.js

```diff
@@ -1,90 +1,123 @@
 import refParser from 'json-schema-ref-parser'
 import { getSecurityDefinitions, getUISecurity } from './securityParser'
+import { getNavigationMethod, getServicesMethod } from './navigationParser'
 import getUIReadySchema from '../schemaParser'
 
 /**
  * Construct navigation and services ready to be consumed by the UI
  *
+ * @param {Object} paths
+ * @param {Array} apiSecurity
+ * @param {Object} securityDefinitions
+ * @return {{navigation: [], services: []}}
+ */
+function getUINavigationAndServices ({ paths, apiSecurity = [], securityDefinitions }) {
+  const { navigationMethods: navigation, servicesMethods } =
+    buildNavigationAndServices(paths, apiSecurity, securityDefinitions)
+
+  // Need to wrap up the methods to be individual services blocks.
+  // This simplifies the component logic.
+  const services = servicesMethods.map(method => {
+    return {
+      title: method.title,
+      methods: [ method ]
+    }
+  })
+
+  return {navigation, services}
+}
+
+/**
+ * Construct navigation and services ready to be consumed by the UI using tags.
+ * This will group the paths by the logical tags.
+ *
  * @param {Array} tags
  * @param {Object} paths
  * @param {Array} apiSecurity
  * @param {Object} securityDefinitions
  * @param {Function} sortFunc
  * @return {{navigation: [], services: []}}
  */
-function getUINavigationAndServices ({ tags, paths, apiSecurity = [], securityDefinitions, sortFunc }) {
+function getUINavigationAndServicesByTags ({ tags, paths, apiSecurity = [], securityDefinitions, sortFunc }) {
   const navigation = []
   const services = []
+  const isFunc = typeof sortFunc === 'function'
 
-  for (let i = 0; i < tags.length; i++) {
+  for (let i = 0, tagLength = tags.length; i < tagLength; i++) {
```

Comment:
I just want to add, it would be nice to integrate benchmarks in our tests, for regressions in the parser (or not-regressions!).

Below is an example benchmark test:

```ts
export function benchmark (timeout, fn) {
  const timeoutAt = Date.now() + timeout
  let iterations = 0

  do {
    fn()
    ++iterations
  } while (Date.now() <= timeoutAt)

  return iterations
}

it('benchmark', () => {
  const base = 2000 // Keep updating this manually, but for CI?
  // In future you'd want to import the previous stable version as a dev dep (maybe git commit hash?) and test against that

  const current = benchmark(500, () => {
    doSomeParsing('foo')
  })

  const percentDiff = Math.round(((current - base) / base) * 100)

  console.dir(`doSomeParsing: ${base} -> ${current} (%${percentDiff})`)

  expect(percentDiff).toBeGreaterThan(20) // It must be 20 percent faster
}
```

In theory iterations will scale across systems, but the `baseline` will be awkward.